src/utils/wallet.ts

```diff
@@ -0,0 +1,225 @@
+/**
+ * Copyright (c) Hathor Labs and its affiliates.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+
+import { crypto, HDPublicKey, HDPrivateKey, Address } from 'bitcore-lib';
+import Mnemonic from 'bitcore-mnemonic';
+import { HD_WALLET_ENTROPY } from '../constants';
+import { XPubError, InvalidWords, UncompressedPubKeyError } from '../errors';
+import Network from '../models/network';
+import _ from 'lodash';
+
+
+const wallet = {
+  /**
+   * Verify if words passed to generate wallet are valid. In case of invalid, returns message
+   *
+   * @param {string} words Words (separated by space) to generate the HD Wallet seed
+   *
+   * @return {Object} {'valid': boolean, 'invalidWords': Array[string], 'words': string} where 'words' is a cleaned
+   * string with the words separated by a single space and invalidWords is an array of invalid words
+   * @throws {InvalidWords} In case the words string is invalid
+   *
+   * @memberof Wallet
+   * @inner
+   */
+  wordsValid(words: string): {valid: boolean, invalidWords?: string[], words?: string} {
+    let newWordsString = '';
+    if (_.isString(words)) {
+      // 1. Replace all non ascii chars by a single space
+      // 2. Remove one or more spaces (or line breaks) before and after the 24 words
+      // 3. Set text to lower case
+      newWordsString = words.replace(/[^A-Za-z0-9]/g, ' ').replace(/\s+/g, ' ').trim().toLowerCase();
+      const wordsArray = newWordsString.split(' ');
+      if (wordsArray.length !== 24) {
+        // Must have 24 words
+        throw new InvalidWords('Must have 24 words.')
+      } else if (!Mnemonic.isValid(newWordsString)) {
+        // Check if there is a word that does not belong to the list of possible words
+        const wordlist = Mnemonic.Words.ENGLISH;
+        const errorList: string[] = [];
+
+        for (const word of wordsArray) {
+          if (wordlist.indexOf(word) < 0) {
+            errorList.push(word);
+          }
+        }
+
+        let errorMessage = '';
+        if (errorList.length > 0) {
+          return {'valid': false, 'invalidWords': errorList};
+        } else {
+          // Invalid sequence of words
+          throw new InvalidWords('Invalid sequence of words.')
```

Comment:
It seems a bug to raise this exception here. I feel that `errorList.length == 0` means that no error has been found, isn't it?

Reply:
Actually not. This code block is inside `!Mnemonic.isValid(newWordsString)`, and the `errorList.length == 0` means that all words are valid in the mnemonic dictionary but they are in the wrong order. That's why the error is 'Invalid sequence of words.'.

Reply:
I got it. Thanks for the explanation.